src/telemetry_subscriber.rs

```diff
@@ -15,74 +17,95 @@ thread_local! {
     static CURRENT_SPAN: RefCell<Vec<Id>> = RefCell::new(vec!());
 }
 
+// TODO: debug impl?
 pub struct TelemetrySubscriber {
-    telemetry_cap: Box<dyn TelemetryCap + Send + Sync + 'static>,
-    spans: CHashMap<Id, RefCt<SpanData>>,
+    telemetry: Box<dyn Telemetry + Send + Sync + 'static>,
     service_name: String,
-    log_level: tracing::Level, // minimum log level
+    spans: Arc<Slab<RwLock<RefCt<SpanData>>>>,
```

Comment:
the `Arc` around the slab shouldn't be necessary; there is only one instance of the subscriber which is itself `Arc`ed.